snuba/web/db_query.py

```diff
@@ -388,6 +388,37 @@ def get_query_cache_key(formatted_query: FormattedQuery) -> str:
     return md5(force_bytes(formatted_query.get_sql())).hexdigest()
 
 
+def check_sorted_sql_key_in_cache(
+    formatted_query_sorted: Optional[FormattedQuery], reader: Reader
+) -> None:
+    cache_partition = _get_cache_partition(reader)
+    if formatted_query_sorted:
+        key_sorted = get_query_cache_key(formatted_query_sorted)
+        percentage = state.get_config(
+            "sort_sql_fields_and_conditions_rollout_percentage", 0
+        )
+        assert isinstance(percentage, int)
+        if percentage > hash_to_probability(key_sorted):
+            key_sorted_with_prefix = f"sorted:{key_sorted}"
+            result = cache_partition.get(key_sorted_with_prefix)
+            if result is not None:
+                write_query_cache_hit_metric("sorted", reader, key_sorted)
```

Comment:
This shouldn't happen here. There should be one single call to the metrics, after both sorted and unsorted caches have been checked. We want to look at the intersection between the two results, and if you log them separately that won't work.

I would change this function to return the results of the cache check instead.

Reply:
Updated as suggested!